### DISCUSSION TITLE
Prevent fallback to unsafe connection when PK mismatch

### DISCUSSION THREAD

**Discussion by lucasteinke** [rustdesk/rustdesk#6527] [General]
### Bug Description

Rustdesk client falls back to an insecure connection when it detects a private key mismatch:

https://github.com/rustdesk/rustdesk/blob/0c4e6f0a8c404e7c1da6d9541fd580c6bd301cd5/src/client.rs#L579-L583

IMHO this may not be done automatically. A MITM attack is easier due to a downgrade attack.

### How to Reproduce

Corrupt the private key on connection, probably? Didn't test if this was a feasible attack since you'd need to use mitmproxy or a patched server which always returns bad private keys.

### Expected Behavior

It may show a dialog like in Firefox to try the insecure connection with an explicit dropdown where you have to confirm that this is the case.

This may not be logged as "info" but maybe as "warning" if it's really a security issue.

### Operating system(s) on local side and remote side

irrelevant

### RustDesk Version(s) on local side and remote side

1.2.3

### Screenshots

This is how Firefox does it:

![image](https://github.com/rustdesk/rustdesk/assets/43927820/e88f8ce7-c97b-477f-85b1-a62eca34ef75)

### Additional Context

Maybe I understood the line of code wrong and this is not a problem at all. In that case, please explain how this is used, possibly in a code comment.

**Comment by rustdesk** [comment]
@sahilyeole please add a security dialog for this (all insecure not-direct-ip connection for flutter only)

- [ ] only send login message or pop out login dialog (depends on if password saved) after user confirm.
- [ ] test reconnection please.
